### PR TITLE
feat(asset): fix [object Object] error message

### PR DESCRIPTION
**Pull request by Catalin-web**
# Pull Request

## 🤨 Rationale

Task: [Don't throw [object Object] if request failed](https://ni.visualstudio.com/DevCentral/_backlogs/backlog/ASW%20SystemLink%20Asset%20Mgmt/Features/?workitem=2905424)

## 👩‍💻 Implementation

Check if the error is fetch error, and format the error

## 🧪 Testing
calibration forecast object object

## ✅ Checklist

- [x] This PR has a title that follows the [commit message format](https://github.com/ni/systemlink-grafana-plugins#commit-message-format).

### CODE REVIEW

**Review by MileaRobertStefan — APPROVED** [review]

File: src/datasources/asset-calibration/AssetCalibrationDataSource.ts [hunk 1]
```diff
@@ -239,6 +239,18 @@ export class AssetCalibrationDataSource extends DataSourceBase<AssetCalibrationQ
       let response = await this.post<CalibrationForecastResponse>(this.baseUrl + '/assets/calibration-forecast', data);
       return response;
     } catch (error) {
+      if (isFetchError(error))
+      {
+        const fetchError = error as FetchError;
+        const statusCode = fetchError.status;
+        const data = fetchError.data;
+        if (data.error && data.error.message) {
```
Comment (saradei-ni) [inline src/datasources/asset-calibration/AssetCalibrationDataSource.ts:247]:
```
const errorMessage = data.error?.message || JSON.stringify(data);
throw new Error(`Query calibration forecast failed with status code: ${statusCode}. Error message: ${errorMessage}`);
```

Suggestion

**Review by saradei-ni — APPROVED** [review]

**Review by kkerezsi — APPROVED** [review]

File: src/datasources/asset-calibration/AssetCalibrationDataSource.ts [hunk 2]
```diff
@@ -239,6 +239,18 @@ export class AssetCalibrationDataSource extends DataSourceBase<AssetCalibrationQ
       let response = await this.post<CalibrationForecastResponse>(this.baseUrl + '/assets/calibration-forecast', data);
       return response;
     } catch (error) {
+      if (isFetchError(error))
```
Comment (kkerezsi) [inline src/datasources/asset-calibration/AssetCalibrationDataSource.ts:242]: @Catalin-web can we have a generic solution also for the base datasource class

Comment (kkerezsi) [reply src/datasources/asset-calibration/AssetCalibrationDataSource.ts:242]: This will be likely an issue in all plugins

**Review by cameronwaterman — APPROVED** [review]

**Pull request merged**